### PR TITLE
[HDInsight] Support new feature:  create kafka cluster with kafka rest proxy

### PR DESCRIPTION
**Pull request by aim-for-better**
This PR is related  with swagger PR [[HDInsight] Support kafka rest proxy feature](https://github.com/Azure/azure-rest-api-specs/pull/7844) 
 1. Support new feature:  create kafka cluster with kafka rest proxy
 2. update azure-mgmt-hdinsight version from 1.1.0 to 1.3.0 
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by aim-for-better** [comment]
@Juliehzl  Could you please help us review this PR? Can we join S162? I added  S162 milestone by myself.

**Comment by aim-for-better** [comment]
@haroldrandom  @Juliehzl  Is there someone can help review our PR? It's already 3 work days since we submitted the PR.

**Comment by haroldrandom** [comment]
Bump SDK version need to updating tests I think?

**Comment by idear1203** [comment]
@haroldrandom The SDK is backward compatible. The newer SDK only adds a few properties to support this new feature. So in general it is not necessary to update tests due to SDK version change.

File: src/azure-cli/azure/cli/command_modules/hdinsight/_params.py [hunk 1]
```diff
@@ -161,6 +164,12 @@ def load_arguments(self, _):
         c.argument('encryption_algorithm', arg_type=get_enum_type(JsonWebKeyEncryptionAlgorithm),
                    arg_group='Customer Managed Key', help='Algorithm identifier for encryption.')
 
+        # Kafka Rest Proxy
+        c.argument('kafka_client_group_id', arg_group='Kafka Rest Proxy',
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/hdinsight/_params.py]: Is it possible to provide short name for your argument here?

File: src/azure-cli/azure/cli/command_modules/hdinsight/_params.py [hunk 2]
```diff
@@ -161,6 +164,12 @@ def load_arguments(self, _):
...
+                   help='The client AAD security group id for Kafka Rest Proxy')
+        c.argument('kafka_client_group_name', arg_group='Kafka Rest Proxy',
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/hdinsight/_params.py]: The same

File: src/azure-cli/HISTORY.rst [hunk 3]
```diff
@@ -15,6 +15,10 @@ Release History
 
 * Remove support for Python 3.4
 
+**HDInsight**
+
+* Support for creating a Kafka cluster with Kafka Rest Proxy
```
Comment (Juliehzl) [inline src/azure-cli/HISTORY.rst]: Declare your package version update here. e.g.
` Upgrade azure-mgmt-hdinsight to 1.3.0`

File: src/azure-cli/azure/cli/command_modules/hdinsight/_params.py [hunk 4]
```diff
@@ -96,8 +96,11 @@ def load_arguments(self, _):
                    help='The size of the data disk in GB, e.g. 1023.')
         c.argument('zookeepernode_size', arg_type=node_size_type)
         c.argument('edgenode_size', arg_type=node_size_type)
+        c.argument('kafka_management_node_size', arg_type=node_size_type)
         c.argument('workernode_count', options_list=['--workernode-count', '-c'], arg_group='Node',
                    help='The number of worker nodes in the cluster.')
+        c.argument('kafka_management_node_count', arg_group='Node',
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/hdinsight/_params.py]: Can you add test for the two new arguments?

**Comment by Juliehzl** [comment]
> @haroldrandom The SDK is backward compatible. The newer SDK only adds a few properties to support this new feature. So in general it is not necessary to update tests due to SDK version change.

We know the newer SDK should be backward compatible, but I think the new SDK may have api version number change, right? Re-recording your tests in hdinsight module based on the new SDK can help double-check the functionality of new SDK and update the request url to make it consistent with the new SDK used. We really recommend you re-record your tests here.

**Comment by Juliehzl** [comment]
LGTM in general. Please resolve all my comments and if CI agrees I will approve.

File: src/azure-cli/azure/cli/command_modules/hdinsight/_params.py (hunk 1 above)
Comment (aim-for-better) [reply src/azure-cli/azure/cli/command_modules/hdinsight/_params.py]: > Is it possible to provide short name for your argument here?

Hi @Juliehzl , Thanks. This is the shortest name that can explain the meaning. And there are already some properties' name are longer than this one. For example : `workernode_data_disk_storage_account_type=None`, `workernode_data_disk_size=None`,
We should consider the meaning firstly, right?

File: src/azure-cli/azure/cli/command_modules/hdinsight/_params.py (hunk 2 above)
Comment (aim-for-better) [reply src/azure-cli/azure/cli/command_modules/hdinsight/_params.py]: > The same

The same with above

File: src/azure-cli/azure/cli/command_modules/hdinsight/_params.py (hunk 1 above)
Comment (Juliehzl) [reply src/azure-cli/azure/cli/command_modules/hdinsight/_params.py]: You can provide `options_list=['--kafka-client-group-id', '-k']` to give both descriptive name and short name. (Feel free to find a vaild short name here)
The same for the following one.

File: src/azure-cli/HISTORY.rst (hunk 3 above)
Comment (aim-for-better) [reply src/azure-cli/HISTORY.rst]: > Declare your package version update here. e.g.
> ` Upgrade azure-mgmt-hdinsight to 1.3.0`

Added

File: src/azure-cli/azure/cli/command_modules/hdinsight/_params.py (hunk 4 above)
Comment (aim-for-better) [reply src/azure-cli/azure/cli/command_modules/hdinsight/_params.py]: > Can you add test for the two new arguments?

Added the test

**Comment by aim-for-better** [comment]
> > @haroldrandom The SDK is backward compatible. The newer SDK only adds a few properties to support this new feature. So in general it is not necessary to update tests due to SDK version change.
> 
> We know the newer SDK should be backward compatible, but I think the new SDK may have api version number change, right? Re-recording your tests in hdinsight module based on the new SDK can help double-check the functionality of new SDK and update the request url to make it consistent with the new SDK used. We really recommend you re-record your tests here.

Hi Zunli, I have added a test for the new properties. Thanks

**Comment by aim-for-better** [comment]
Hi @Juliehzl , how do you like our PR?

File: src/azure-cli/azure/cli/command_modules/hdinsight/tests/latest/test_hdinsight_commands.py [hunk 5]
```diff
@@ -47,6 +47,16 @@ def test_hdinsight_cluster_kafka(self, storage_account_info):
             HDInsightClusterTests._kafka_arguments()
         )
 
+    # Uses 'rg' kwarg
+    @ResourceGroupPreparer(name_prefix='hdicli-', location=location, random_name_length=12)
+    @StorageAccountPreparer(name_prefix='hdicli', location=location, parameter_name='storage_account')
+    def test_hdinsight_cluster_kafka_with_rest_proxy(self, storage_account_info):
+        self._create_hdinsight_cluster(
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/hdinsight/tests/latest/test_hdinsight_commands.py]: Typically we will use test scenarios for commands. Can you add test for your new parameter with self.cmd here?

Comment (aim-for-better) [reply src/azure-cli/azure/cli/command_modules/hdinsight/tests/latest/test_hdinsight_commands.py]: > Typically we will use test scenarios for commands. Can you add test for your new parameter with self.cmd here?

Hi @Juliehzl , Thanks for your comment. We really cover the self.cmd lever.
The self._create_hdinsight_cluster will actually call `az hdinsight create` cmd. 
The definition of `_create_hdinsight_cluster` is at [here](https://github.com/Azure/azure-cli/blob/22035a769df76a48d964477f1c0e0e63ed6c174a/src/azure-cli/azure/cli/command_modules/hdinsight/tests/latest/test_hdinsight_commands.py#L297)

And from this [line](https://github.com/Azure/azure-cli/blob/22035a769df76a48d964477f1c0e0e63ed6c174a/src/azure-cli/azure/cli/command_modules/hdinsight/tests/latest/test_hdinsight_commands.py#L305) we can see that this method wil call `az hdinsight create` cmd.

Thanks

**Pull request merged**